File: src/main/java/assignment2/TypeErrorException.java
package assignment2;

/**
 * Exception thrown to indicate a type error at a specific line in the source code.
 */
public class TypeErrorException extends CompilerException {

    /**
     * Constructs a new TypeErrorException with the specified detail message and line number.
     *
     * @param message the detail message
     * @param line    the line number where the type error occurred
     */
    public TypeErrorException(String message, int line) {
        super(message, line);
    }

    /**
     * Returns a string representation of the type error exception.
     *
     * @return a string describing the type error and its line number
     */
    @Override
    public String toString() {
        return "Type error at line " + getLine() + ": " + getMessage();
    }
}

File: src/main/scala/assignment3/TypeErrorException.scala
package assignment3

/** Exception thrown to indicate a type error at a specific location in the source code (Scala port).
  * @deprecated Use TypeDiag instead. Will be removed in 3.0.
  */
@deprecated("Use TypeDiag instead", "2.0")
class TypeErrorException(message: String, line: Int, column: Int) extends SourcePositionedException(message, line, column) {
  def this(message: String, line: Int) = this(message, line, -1)
}
